common/sky/moon.py

```diff
@@ -0,0 +1,481 @@
+                    - e * 0.000111 * np.cos(M)
+                    + 0.000103 * np.cos(4 * D - 2 * MPR)
+                    - 0.000084 * np.cos(2 * MPR - 2 * D)
+                    - e * 0.000083 * np.cos(2 * D + M)
+                    + 0.000079 * np.cos(2 * D + 2 * MPR)
+                    + 0.000072 * np.cos(4 * D)
+                    + e * 0.000064 * np.cos(2 * D - M + MPR)
+                    - e * 0.000063 * np.cos(2 * D + M - MPR)
+                    + e * 0.000041 * np.cos(M + D)
+                    + e * 0.000035 * np.cos(2 * MPR - M)
+                    - 0.000033 * np.cos(3 * MPR - 2 * D)
+                    - 0.000030 * np.cos(MPR + D)
+                    - 0.000029 * np.cos(2 * F - 2 * D)
+                    - e * 0.000029 * np.cos(2 * MPR + M)
+                    + e * e * 0.000026 * np.cos(2 * D - 2 * M)
+                    - 0.000023 * np.cos(2 * F - 2 * D + MPR)
+                    + e * 0.000019 * np.cos(4 * D - M - MPR))
+
+        self.BETA = Angle(np.deg2rad(beta), unit=u.rad)
+        self.LAMBDA = Angle(np.deg2rad(lambd), unit=u.rad)
+    
+    def low_precision_location(self, obs: EarthLocation) -> Tuple[SkyCoord, float]:
+        """
+        This is the same as the high precision method, but with a different set of coefficients.  
+        The difference is small. Good to about 0.1 deg, from the 1992 Astronomical Almanac, p. D46.
+        Note that input time is a float.
+        """
+
+        self._low_precision_calculations()
+        # Terrestrial time with julian day
+        sid = local_sidereal_time(self.time, obs)
+        lat = obs.lat
+        distance = 1. / np.sin(self.PIE)
+
+        l = np.cos(self.BETA) * np.cos(self.LAMBDA)
+        m = 0.9175 * np.cos(self.BETA) * np.sin(self.LAMBDA) - 0.3978 * np.sin(self.BETA)
+        n = 0.3978 * np.cos(self.BETA) * np.sin(self.LAMBDA) + 0.9175 * np.sin(self.BETA)
+
+        x = l * distance
+        y = m * distance
+        z = n * distance  # /* for topocentric correction */
+  
+        x = x - np.cos(lat) * np.cos(sid)
+        y = y - np.cos(lat) * np.sin(sid)
+        z = z - np.sin(lat)
+
+        topo_dist = np.sqrt(x * x + y * y + z * z)
+
+        l = x / topo_dist
+        m = y / topo_dist
+        n = z / topo_dist
+
+        alpha = np.arctan2(m, l)
+        delta = np.arcsin(n)
+        distancemultiplier = Distance(EQUAT_RAD, unit=u.m)
+
+        fr = current_geocent_frame(self.time)
+        return SkyCoord(alpha, delta, topo_dist * distancemultiplier, frame=fr), topo_dist
+    
+    def accurate_location(self, obs: EarthLocation) -> Tuple[SkyCoord, float]:
+        """  
+        Compute topocentric location and distance of moon to better accuracy.
+
+        This is good to about 0.01 degrees
+
+        Parameters
+        ----------
+        obs : EarthLocation
+            location on earth.
+
+        Returns:
+
+        tuple of a SkyCoord and a distance.
+
+        """
+        self._high_precision_calculations()
```

Comment:
e.g. do we always have to do `self._high_precision_calculations()` if none of the values used in that method have changed? That's the kind of thing I want to check.